forgerock-authenticator/src/main/java/org/forgerock/android/auth/PushNotification.java

```diff
@@ -299,16 +307,79 @@ String serialize() {
     }
 
     /**
-     * Accepts the push authentication request.
-     * @param listener Listener for receiving the HTTP call response code.
+     * Accepts the push authentication request. Use this method to approve notification of
+     * type {@code PushType.DEFAULT}.
+     * @param listener Listener for receiving the authentication result.
      */
     public final void accept(@NonNull FRAListener<Void> listener) {
-        Logger.debug(TAG, "Accept Push Authentication request for message: %s", getMessageId());
-        performAcceptDenyAsync(true, listener);
+        if (this.pushType == PushType.DEFAULT) {
+            Logger.debug(TAG, "Accept Push Authentication request for message: %s", getMessageId());
+            performAcceptDenyAsync(true, listener);
+        } else {
+            listener.onException(new PushMechanismException("Error processing the Push " +
+                    "Authentication request. This method cannot be used to process notification " +
+                    "of type: " + this.pushType));
+        }
+    }
+
+    /**
+     * Accepts the push notification request with the challenge response. Use this method to handle
+     * notification of type {@code PushType.BIOMETRIC}.
```

Comment:
I think this should be PushType.Challenge right?

Reply:
yes, nice catch